src/odemis/gui/model/__init__.py

```diff
@@ -595,8 +602,51 @@ def __init__(self, main):
         # VA for autofocus procedure mode
         self.autofocus_active = BooleanVA(False)
 
+class CryoGUIData(MicroscopyGUIData):
+    """
+    Represents an interface for handling cryo microscopes.
+    """
+    def __init__(self, main):
+        if "cryo" not in main.role:
+            raise ValueError(
+                "Expected a cryo microscope role but found it to be %s." % main.role)
+        MicroscopyGUIData.__init__(self, main)
 
-class LocalizationGUIData(MicroscopyGUIData):
+    def add_new_feature(self, pos_x, pos_y, pos_z=None, f_name=None, milling_angle=DEFAULT_MILLING_ANGLE):
+        """
+        Create a new feature and add it to the features list
+        """
+        if not f_name:
+            existing_names = [f.name.value for f in self.main.features.value]
+            f_name = make_unique_name("Feature-1", existing_names)
+        if pos_z is None:
+            pos_z = self.main.focus.position.value['z']
+        feature = CryoFeature(f_name, pos_x, pos_y, pos_z, milling_angle)
+        self.main.features.value.append(feature)
+        return feature
+
+    # Todo: find the right margin
+    ATOL_FEATURE_POS = 0.1e-3  # m
+
+    def select_current_position_feature(self):
+        """
+        Given current stage position, either select one of the features closest to the position or create a new one with the position
+        :return: (CryoFeature) the feature selected for the current position
+        """
+        current_position = self.main.stage.position.value
+        for feature in self.main.features.value:
+            feature_dist = math.hypot(feature.pos.value[0] - current_position["x"],
+                                      feature.pos.value[1] - current_position["y"])
+            if feature_dist <= self.ATOL_FEATURE_POS:
+                self.currentFeature.value = feature
+                break
+        else:
+            # create new feature if no close feature found
+            feature = self.add_new_feature(current_position["x"], current_position["y"],
+                                                          self.main.focus.position.value["z"])
+            self.main.currentFeature.value = feature
```

Comment:
The block just above uses `self.currentFeature`, while here it's `self.main.currentFeature`. That's fishy.